examples/field_examples/viking_graben_line12/rtm/rtm.jl

```diff
@@ -0,0 +1,188 @@
+d = Tuple(Float32(i) for i in d)
+o = Tuple(Float32(i) for i in o)
+if modeling_type == "scalar"
+    model0 = Model(n, d, o, m0)
+elseif modeling_type == "acoustic"
+    rho0 = rho_from_slowness(m0)
+    model0 = Model(n, d, o, m0, rho=rho0)
+end
+
+x = (o[1]:d[1]:o[1]+(n[1]-1)*d[1])./1000f0
+z = (o[2]:d[2]:o[2]+(n[2]-1)*d[2])./1000f0
+
+global seabed_ind = Int.(round.(seabed./d[2]))
+if modeling_type == "scalar"
+    model0.m[:,1:seabed_ind] .= (1/vwater)^2
+elseif modeling_type == "acoustic"
+    model0.m[:,1:seabed_ind] .= (1/vwater)^2
+    model0.b[:,1:seabed_ind] .= 1f0/rhowater
+end
+
+# Set up wavelet and source vector
+src_geometry = Geometry(container; key = "source", segy_depth_key = segy_depth_key_src)
+
+# sampling frequency
+fs = 1000f0/src_geometry.dt[1]
+# Nyquist frequency
+f_nyq = fs/2f0
+
+# setup wavelet
+wavelet_raw = readdlm(wavelet_file, skipstart=wavelet_skip_start)
+itp = LinearInterpolation(0:wavelet_dt:wavelet_dt*(length(wavelet_raw)-1), wavelet_raw[:,1], extrapolation_bc=0f0)
+wavelet = Matrix{Float32}(undef,src_geometry.nt[1],1)
+wavelet[:,1] = itp(0:src_geometry.dt[1]:src_geometry.t[1])
+q = judiVector(src_geometry, wavelet)
+
+############################################## FWI #################################################
+
+grad_mem = 40 # Based on n and CFL condition
+
+mem = Sys.free_memory()/(1024^3)
+t_sub = max(1, ceil(Int, nworkers()*grad_mem/mem))
+@info "modeling_type: $modeling_type"
+@info "subsampling factor: $t_sub"
+
+# JUDI options
+jopt = JUDI.Options(
+    space_order=16,
+    limit_m = true,
+    buffer_size = buffer_size,
+    optimal_checkpointing=false,
+    IC = "isic",
+    # subsampling_factor=t_sub     # subsampling_factor with space_order set leads to exception (probably bug)
+    )
+
+# Right-hand preconditioners (model topmute)
+idx_wb = find_water_bottom(reshape(model0.m, model0.n))
+Tm = judiTopmute(model0.n, idx_wb, 10)  # Mute water column
+S = judiDepthScaling(model0)
+Mr = S*Tm
+
+# Left-hand side preconditioners
+Ml = judiDataMute(q.geometry, d_obs.geometry, vp=1100f0, t0=0.001f0, mode=:reflection) # keep reflections
+
+# Setup operators
+Pr = judiProjection(d_obs.geometry)
+F = judiModeling(model0; options=jopt)
+Ps = judiProjection(q.geometry)
+J = judiJacobian(Pr*F*adjoint(Ps), q)
+
+shot_from = 1
+shot_to = length(d_obs)
+shot_step = 3   # only compute each 3rd shot (that is enough)
+
+indsrc = rand(shot_from:shot_from+shot_step-1):shot_step:shot_to
+
+# Topmute
+d_obs = Ml[indsrc]*d_obs[indsrc]
+
+# RTM
+rtm = adjoint(Mr)*adjoint(J[indsrc])*d_obs
+
+rtm_plt = Plots.heatmap(x, z, adjoint(reshape(rtm, model0.n[1], model0.n[2])), c=:bluesreds, 
+    xlims=(x[1],x[end]), 
+    ylims=(z[1],z[end]), yflip=true,
+    clim=(-maximum(rtm)/3f0, maximum(rtm)/3f0),
+    title="RTM",
+    xlabel="Lateral position [km]",
+    ylabel="Depth [km]",
+    dpi=600)
+Plots.savefig(rtm_plt, dir_out * "RTM_$(modeling_type).png")
+
+h5open(dir_out * "rtm_$(modeling_type).h5", "w") do file
```

Comment:
could wrap this into a `write_model(file, name, model)` to be reused everywhere since the rest is mostly properties from `model`

Reply:
done